packages/compose/src/hooks/use-debounce/index.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * useDebounce hook
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Use to debounce a value that is changing too frequently.
+ *
+ * Example usage:
+ *
+ * ```js
+ *	// An example using the useDebounce hook with useResizeObserver()
+ *  // to debounce a rapidly changing width value
+ *
+ *	const [ resizeListener, sizes ] = useResizeObserver();
+ *
+ *	useDebounce( ( sz ) => {
+ *
+ *		// ... do your thing here ...
+ *
+ *	}, 200, sizes.width );
+ *
+ *	return (
+ *		<div>
+ *			{resizeListener}
+ *		</div>
+ *	);
+ *	```
+ *
+ * @param {Function} callback The function to call with the debounced value.
+ * @param {number}   delay    The amount to delay time in ms.
+ * @param {*}        deps     The dependent value changing you want to debounce.
+ *
+ */
+const useDebounce = ( callback, delay, deps ) => {
```

Comment:
Could be an interesting question: How would we (or would we) want to deal with a changing `delay` value.

It might be fine enough to include in the `deps` of the `useEffect`. Or just choose not to support this usage.

Reply:
I could see maybe if you wanted to do an escalating delay, but we can see if that need or request would come up.